### DISCUSSION TITLE
Adding comments with signature using `:ab` and `:iabbrev`

### DISCUSSION THREAD

**Discussion by rameshsanth** [numToStr/Comment.nvim#283] [Q&A] [answered]
This belongs to both Ideas and Q&A section

I feel this can be a very useful feature if integrated with this plugin.

I have below vi commands to insert comments with signature in them.. 
```
au filetype h,c,cpp iabbr msb <Home>/* --my_signature: changes begin <C-R>=strftime("%c")<CR> */
au filetype h,c,cpp iabbr mse <Home>/* --my_signature: changes end <C-R>=strftime("%c")<CR> */
au filetype h,c,cpp iabbr dbg // --my_signature debug
au filetype h,c,cpp iabbr tbd // --my_signature TODO
au fileType h,c,cpp inoremap <buffer> cmt /* --my_signature */<esc>2hi
```

As you can see
msb/mse adds "Start"/"End" with timestamp
dbg adds signature with debug 
and so on..

The issue is, this set needs to be replicated for each type of file. C/Python/Shell/ etc.. Plus this is VIM commands.

Is there a way to do this programmatically using the Comment plugin ?

I started with something as below which adds comment at the end of the line.. but I don't know how to tie this with "iabbr"
I'm not a Lua expert. I would appreciate help to complete this 

```lua
-- Add comment at end of the line with "signature"
local api = require('Comment.api')
local config = require('Comment.config'):get()

local signature = {"--my_signature"}
api.insert.linewise.eol(config)
vim.api.nvim_put(signature, 'c', true, true)
```

**Comment by numToStr** [comment, marked as answer]
I am not that familiar with how abbreviations family of commands works but I quickly checked, and I believe this is what you want:

```lua
local api = require("Comment.api")

-- Add comment at end of the line with "signature"
function _G.Sign()
    api.insert.linewise.eol()
    vim.api.nvim_put({ "--my_signature" }, "c", true, true)
end

vim.cmd("iabbr msb <C-o>:lua Sign()<CR>")
```

> NOTE - Currently there is a bug with `eol` where `a` is inserted at the end.

**Reply by rameshsanth** [reply]
Yes, this works except for that "a" that gets appended. Is it bug in Comment.nvim ?

**Reply by numToStr** [reply]
Yes, sadly it is a bug 😅
